galite-core/src/main/kotlin/org/kopi/galite/visual/ApplicationContext.kt

```diff
@@ -18,15 +18,323 @@
+   */
+  abstract fun isWebApplicationContext(): Boolean
+
   companion object {
-    fun getDBContext(): DBContext {
-      TODO()
+
+    //-----------------------------------------------------------
+    // UTILS
+    //-----------------------------------------------------------
+
+    /**
+     * Returns the default configuration of the Application
+     */
+    fun getDefaults(): ApplicationConfiguration = applicationContext.getApplication().getApplicationConfiguration()
+
+    /**
+     * Returns the [Application] menu.
+     * @return The [Application] menu.
+     */
+    fun getMenu(): VMenuTree = applicationContext.getApplication().getMenu()
+
+    /**
+     * Returns the [LocalizationManager] instance.
+     * @return The [LocalizationManager] instance.
+     */
+    fun getLocalizationManager(): LocalizationManager = applicationContext.getApplication().getLocalizationManager()
+
+    /**
+     * Returns the default application [Locale].
+     * @return The default application [Locale].
+     */
+    fun getDefaultLocale(): Locale = applicationContext.getApplication().getDefaultLocale()
+
+    /**
+     * Returns the application [Registry].
+     * @return the application [Registry].
+     */
+    fun getRegistry(): Registry = applicationContext.getApplication().getRegistry()
+
+    /**
+     * Returns the application [DBContext].
+     * @return The application [DBContext].
+     */
+    fun getDBContext(): DBContext = applicationContext.getApplication().getDBContext()
+
+    /**
+     * Returns `true` if the [Application] should only generate help.
+     * @return `true` if the [Application] should only generate help.
+     */
+    fun isGeneratingHelp(): Boolean = applicationContext.getApplication().isGeneratingHelp()
+
+    /**
+     * Displays an error message outside a model context. This can happen when launching a module.
+     * @param parent The parent component.
+     * @param message The message to be displayed.
+     */
+    fun displayError(parent: UComponent, message: String) {
+      applicationContext.getApplication().displayError(parent, message)
+    }
+
+    // ---------------------------------------------------------------------
+    // SEND A BUG REPORT
+    // ---------------------------------------------------------------------
+
+    /**
+     * Reports a trouble at execution time.
+     *
+     * @param     module          the module where the trouble was detected
+     * @param     reason          the exception that triggered the bug report
+     */
+    fun reportTrouble(module: String,
+                      place: String,
+                      data: String,
+                      reason: Throwable) {
+
+      when {
+        applicationContext.getApplication().isNoBugReport() -> {
+          println("notice: reporting trouble is disabled, no mail will be sent.")
+          System.err.println(reason.message)
+          reason.printStackTrace(System.err)
+        }
+        else -> {
```

Comment:
you still using when statement whereas you have two conditions: Can simply done with if..else statement